Created GasStation.cs class

CCianfloneDev opened the pull request.

CCianfloneDev committed:
Create GasStation.cs

CCianfloneDev committed:
Add files via upload

CCianfloneDev committed:
Add files via upload

DZ-FSDev commented on ColesStopAndShop/GasStation.cs:
```diff
@@ -0,0 +1,96 @@
+        public int NumberOfEmployees
+        {
+            get;
+            set;
```
- [x] Add constraint to prevent negative employees.

DZ-FSDev commented on ColesStopAndShop/GasStation.cs:
```diff
@@ -0,0 +1,96 @@
+        /// <param name="isRestStop">Is a rest stop or not.</param>
+        public GasStation(string address, int numberOfGasPumps, int storeId, int numberOfEmployees, bool servesHotFood, bool isRestStop)
+        {
+            Address = address;
+            NumberOfGasPumps = numberOfGasPumps;
+            StoreId = storeId;
+            NumberOfEmployees = numberOfEmployees;
```
- [x] Add exception checking to maintain data integrity.

DZ-FSDev commented on ColesStopAndShop/GasStation.cs:
```diff
@@ -0,0 +1,96 @@
+        /// <returns></returns>
+        public override string ToString()
+        {
+            return $"Gas station:" +
```
- [x] There is no need to interpolate this first string fragment.

DZ-FSDev commented on ColesStopAndShop/GasStation.cs:
```diff
@@ -0,0 +1,96 @@
+        public override string ToString()
+        {
+            return $"Gas station:" +
+                   $"\n Address: {Address}" +
+                   $"\n Number of gas pumps: {NumberOfGasPumps}" +
+                   $"\n Store ID: {StoreId}" +
+                   $"\n Number of employees: {NumberOfEmployees}" +
+                   $"\n Has hot food?: {ServesHotFood}" +
+                   $"\n Is a rest stop?: {IsRestStop}";
```
- [x] Consider placing new-line chars at the end of string fragments.

DZ-FSDev reviewed (changes requested):
**CR for #23**

CCianfloneDev committed:
Add files via upload

CCianfloneDev committed:
Update ItemId.cs

added more items + documentation.

CCianfloneDev committed:
Merge pull request #25 from CCianfloneDev/CCianfloneDev-ItemId-enum

Added  ItemId.cs enum

CCianfloneDev committed:
Merge pull request #28 from CCianfloneDev/CCianfloneDev-program-class

Added program.cs

CCianfloneDev committed:
Update CashRegister.cs

Added cashBalance field.

CCianfloneDev committed:
Update CashRegister.cs

Corrected documentation for creditBalance.

CCianfloneDev committed:
Update CashRegister.cs

Added exception checking for CreditBalance.

CCianfloneDev committed:
Update CashRegister.cs

Added exception checking for DebitBalance.

CCianfloneDev committed:
Update CashRegister.cs

Added GST and PST rate. Kept SalesTaxRate prop but it now combines PST and GST for easier total calculation within PaymentProcess.

CCianfloneDev committed:
Update CashRegister.cs

Removed association to store via a storeId. It now is associated to a store via a reference to a constructed GasStation.

CCianfloneDev committed:
Update CashRegister.cs

Added exception checking within constructor.

CCianfloneDev committed:
Update CashRegister.cs

Changed prices from int to decimal

CCianfloneDev committed:
Created PaymentType.cs enum

CCianfloneDev committed:
Update CashRegister.cs

Corrected logic problem with 2d loop. 
Cash register no longer requires cashier to enter in price of item, the cashier enters in itemId (mimics scanning barcode) and it is used to reference the price of that item which is defined in a dict within the gas station.

CCianfloneDev committed:
Update GasStation.cs

CCianfloneDev committed:
Update Program.cs

CCianfloneDev committed:
Merge pull request #24 from CCianfloneDev/CCianfloneDev-CashRegister-class

Added CashRegister.cs class

CCianfloneDev committed:
Update GasStation.cs

Added exception checking in some props.
Added fields

CCianfloneDev committed:
Update GasStation.cs

Added exception checking within constructor.

CCianfloneDev committed:
Update GasStation.cs

Cleaned up ToString override method.
